src/main/java/org/terasology/gooeysQuests/quests/mines/MinesQuestSystem.java

```diff
@@ -0,0 +1,202 @@
+package org.terasology.gooeysQuests.quests.mines;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityBuilder;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeDeactivateComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.gooeysQuests.api.*;
```

Comment:
Avoid wildcard import - should be easily Google-able instructions for how to avoid them in IntelliJ (and _why_ we avoid them, if you're curious)